### DISCUSSION TITLE
Import vue-router in dynamically loaded vue component

### DISCUSSION THREAD

**Discussion by Remo** [FranckFreiburger/vue3-sfc-loader#211] [Q&A]
I managed to get my routing up and running based on this https://github.com/FranckFreiburger/vue3-sfc-loader/discussions/172.

I then tried to do some magic in a dynamically loaded component.
I can import things from vue with `import { ref, onMounted } from 'vue'`, but as soon as I try to load something from vue-router like `import { useRoute } from 'vue-router'`, it fails.

In the end, I'm trying to access some route params from within onMounted.

Here's an example:
https://jsbin.com/nidexuqero/edit?html,output

**Comment by Remo** [comment]
I think I found a way, but I'm not sure this is the best way to go:

```
<script setup>
    import { onMounted } from 'vue'

    const router = window.VueRouter.useRouter()
    const route = window.VueRouter.useRoute()

    onMounted(async () => {
        await router.isReady()
        if (route.params && route.params.form) {
            window.alert(route.params.form)
        }
    })
</script>
```

**Comment by FranckFreiburger** [comment]
Hello Remo,

my bad, the example is incorrect.
In moduleCache, VueRouter must be stored as 'vue-router', and not 'VueRouter'

fix:
```
    const options = {
      moduleCache: { vue: Vue, 'vue-router': VueRouter },
```

(https://github.com/FranckFreiburger/vue3-sfc-loader/discussions/172 is updated)

**Comment by Remo** [comment]
Now I understand how `moduleCache` works! Thanks a lot of your quick response, I appreciate it.
